src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/ReviewCodeForXssVulnerabilitiesTests.cs

```diff
@@ -213,5 +213,55 @@ protected void Page_Load(object sender, EventArgs e)
                 },
             }.RunAsync();
         }
+
+        [Fact]
+        public async Task HttpServerUtility_HtmlEncode_StringWriterOverload_NoDiagnostic()
+        {
+            const string code = @"
+using System;
+using System.Data.SqlClient;
+using System.IO;
+using System.Web;
+
+public partial class WebForm : System.Web.UI.Page
+{
+    protected void Page_Load(object sender, EventArgs e)
+    {
+        string input = Request.Form[""in""];
+        StringWriter w = new StringWriter();
+        Server.HtmlEncode(input, w);
+        Response.Write(""<HTML>"" + w.ToString() + ""</HTML>"");
+        // make sure it is not like any unknown method breaks the taint path, it should warn about sql injection
+        SqlCommand sqlCommand = new SqlCommand(w.ToString());
+    }
+}";
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultForTaintedDataAnalysis,
+                TestState =
+                {
+                    Sources =
+                    {
+                        code,
+                    }
+                },
+            }.RunAsync();
+
+            await new CSharpSecurityCodeFixVerifier<Microsoft.NetCore.Analyzers.Security.ReviewCodeForSqlInjectionVulnerabilities, Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>.Test
```

Comment:
Can this be put in `ReviewCodeForCommandExecutionVulnerabilitiesTests`? And the `code` could be a constant shared by both test classes.